### PR TITLE
Add function to compute latest persistence forecasts

### LINKED ISSUE #375: precomputed persistence forecasts for reference sites
We should to supply a couple of persistence forecasts for reference sites. They should be recomputed when new data is fetched.

I suggest the following configurations:

```
forecast_75_60 = datamodel.Forecast(
    name=name,
    site=site,
    variable=variable,
    interval_label='beginning',
    interval_value_type='mean',
    interval_length='1h',
    issue_time_of_day=datetime.time(hour=0, minute=45),
    run_length=pd.Timedelta('1h'),
    lead_time_to_start=pd.Timedelta('75min'),
    forecast_id='',
    extra_parameters='',
)

forecast_5_5 = datamodel.Forecast(
    name=name,
    site=site,
    variable=variable,
    interval_label='beginning',
    interval_value_type='mean',
    interval_length='5min',
    issue_time_of_day=datetime.time(hour=0),
    run_length=pd.Timedelta('5min'),
    lead_time_to_start=pd.Timedelta('5min'),
    forecast_id='',
    extra_parameters='',
)

forecast_15_15 = datamodel.Forecast(
    name=name,
    site=site,
    variable=variable,
    interval_label='beginning',
    interval_value_type='mean',
    interval_length='15min',
    issue_time_of_day=datetime.time(hour=0),
    run_length=pd.Timedelta('15min'),
    lead_time_to_start=pd.Timedelta('15min'),
    forecast_id='',
    extra_parameters='',
)
```

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #375  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

which can later be added as a periodic task to the API to generate reference persistence forecasts

### CODE REVIEW

File: solarforecastarbiter/io/reference_observations/common.py [hunk 1]
```diff
@@ -426,15 +425,17 @@ def create_one_forecast(api, site, template_forecast, variable,
         to extra parameters.
     variable : string
         Variable measured in the forecast.
+    **extra_params
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/common.py:428]:
```suggestion
    **extra_params : dict
```

File: solarforecastarbiter/io/reference_observations/common.py [hunk 2]
```diff
@@ -535,3 +541,83 @@ def create_forecasts(api, site, variables, templates):
                 create_one_forecast(api, site, template_fx, var,
                                     piggyback_on=piggyback_on))
     return created
+
+
+def create_persistence_forecasts(api, site, variables, templates):
+    """Create persistence Forecast objects for each Observation at the
+    ``site`` with variable in ``variables``. Each Forecast in templates
+5    will be updated with the appropriate parameters for each variable.
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/common.py:549]:
```suggestion
     will be updated with the appropriate parameters for each variable.
```

File: solarforecastarbiter/io/reference_observations/common.py [hunk 3]
```diff
@@ -535,3 +541,83 @@ def create_forecasts(api, site, variables, templates):
...
+            created.append(fx_id)
+    return created
+
+
+def create_forecasts(api, site, variables, templates):
+    """Create Forecast objects (NWP based and persistence) for each of
+    variables. Each Forecast in templates will be updated with the
+    appropriate parameters for each variable.
+
+    Templates with the 'is_reference_persistence_forecast' key  in
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/common.py:584]:
```suggestion
    Templates with the 'is_reference_persistence_forecast' key in
```

File: solarforecastarbiter/io/reference_observations/common.py [hunk 4]
```diff
@@ -535,3 +541,83 @@ def create_forecasts(api, site, variables, templates):
...
+    -------
+    list
+        A list of all Forecast/ProbabilisticForecast objects created
+
+    See Also
+    --------
+    :py:func:`solarforecastarbiter.io.reference_data.common.create_nwp_forecasts
+    :py:func:`solarforecastarbiter.io.reference_data.common.create_persistence_forecasts
+
+    """  # NOQA
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/common.py:612]:
```suggestion
    """  # NOQA: E501
```

File: solarforecastarbiter/io/reference_observations/default_forecasts.py [hunk 5]
```diff
@@ -112,5 +112,57 @@ def is_in_nwp_domain(site):
 ]
 
 
+TEMPLATE_DETERMINISTIC_PERSISTENCE_FORECASTS = [
+    Forecast(
+        name='Hour Ahead Persistence',
+        issue_time_of_day=dt.time(0, 45),
+        lead_time_to_start=pd.Timedelta('75min'),
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/default_forecasts.py:119]:
```suggestion
        issue_time_of_day=dt.time(0, 0),
        lead_time_to_start=pd.Timedelta('60min'),
```

Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/default_forecasts.py:119]: I think we should keep it simple for the standard reference forecasts.

File: solarforecastarbiter/reference_forecasts/main.py [hunk 6]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+    generator of (Forecast, Observation, next_issue_time, index)
+
+    """
+    user_info = session.get_user_info()
+    observation_dict = {obs.observation_id: obs for obs in observations}
+    for fx in forecasts:
+        if not _is_reference_persistence_forecast(fx.extra_parameters):
+            logger.debug(
+                'Forecast %s is not labeled as a reference persistence forecast',  # NOQA
+                fx.forecast_id)
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:509]:
```suggestion
                'Forecast %s is not labeled as a reference '
                'persistence forecast',  fx.forecast_id)
```

File: solarforecastarbiter/reference_forecasts/main.py [hunk 7]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+            logger.warning(
+                'Failed to decode extra_parameters for %s: %s as JSON',
+                fx.name, fx.forecast_id)
+            continue
+
+        try:
+            observation_id = extra_parameters['observation_id']
+        except KeyError:
+            logger.error(
+                ('Forecast, %s: %s, has no observation_id to base forecasts'
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:530]: I'm not positive but I don't think you need the `()` around the strings here or below because they're already encapsulated in the function call's `()`.

File: solarforecastarbiter/reference_forecasts/main.py [hunk 8]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+            # data before deciding to make a persistence fx
+            next_issue_time = utils.get_next_issue_time(
+                fx, obs_maxt - fx.run_length)
+        else:
+            next_issue_time = utils.find_next_issue_time_from_last_forecast(
+                fx, fx_maxt)
+
+        # now find all the run times that can be made based on the
+        # last observation timestamp
+        while next_issue_time <= max_run_time:
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:566]: Can this be a for loop over a DatetimeIndex? It's not immediately obvious that the while loop is guaranteed to exit.

File: solarforecastarbiter/reference_forecasts/utils.py [hunk 9]
```diff
@@ -100,6 +100,33 @@ def get_forecast_start_end(forecast, issue_time,
...
+    *last_forecast_time* with a full length forecast.
+
+    Parameters
+    ----------
+    forecast : datamodel.Forecast
+    last_forecast_time : pd.Timestamp
+        Last timestamp avaible for the forecast
+
+    Returns
+    pd.Timestamp
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/utils.py:117]:
```suggestion
    Returns
    -------
    pd.Timestamp
```

File: solarforecastarbiter/reference_forecasts/utils.py [hunk 10]
```diff
@@ -100,6 +100,33 @@ def get_forecast_start_end(forecast, issue_time,
...
+    Find the next issue time for *forecast* based on the timestamp of the
+    last forecast value. If *last_forecast_time* is not the end of a forecast
+    run, the issue time returned will be the issue time that overwrites
+    *last_forecast_time* with a full length forecast.
+
+    Parameters
+    ----------
+    forecast : datamodel.Forecast
+    last_forecast_time : pd.Timestamp
+        Last timestamp avaible for the forecast
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/utils.py:114]:
```suggestion
        Last timestamp available for the forecast
```

File: solarforecastarbiter/reference_forecasts/utils.py [hunk 11]
```diff
@@ -100,6 +100,33 @@ def get_forecast_start_end(forecast, issue_time,
     return forecast_start, forecast_end
 
 
+def find_next_issue_time_from_last_forecast(forecast,  last_forecast_time):
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/utils.py:103]:
```suggestion
def find_next_issue_time_from_last_forecast(forecast, last_forecast_time):
```

File: solarforecastarbiter/reference_forecasts/utils.py [hunk 12]
```diff
@@ -100,6 +100,33 @@ def get_forecast_start_end(forecast, issue_time,
...
+
+    Returns
+    pd.Timestamp
+        The next issue time for the forecast
+    """
+    last_probable_issue_time = (last_forecast_time -
+                                forecast.run_length -
+                                forecast.lead_time_to_start)
+    if forecast.interval_label != 'ending':
+        last_probable_issue_time += forecast.interval_length
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/utils.py:124]:
```suggestion
    # for beginning & instantaneous labels, last_probable_issue_time
    # is currently e.g. 13:55 - 60 min - 0 min = 12:55, so add the 
    # interval_length to make last_probable_issue_time = 13:00
    if forecast.interval_label != 'ending':
        last_probable_issue_time += forecast.interval_length
```

File: solarforecastarbiter/io/reference_observations/default_forecasts.py [hunk 13]
```diff
@@ -112,5 +112,57 @@ def is_in_nwp_domain(site):
 ]
 
 
+TEMPLATE_DETERMINISTIC_PERSISTENCE_FORECASTS = [
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/default_forecasts.py:115]: How about a day ahead persistence with hour-ending intervals?

File: solarforecastarbiter/reference_forecasts/main.py (hunk 8 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:566]: hmm, we should rely on ``utils.get_next_issue_time``/``utils.get_issue_times`` so we would first need a similar loop to construct the issue times. The nice thing here is that it is a generator instead of building a list and looping through it. I think if one trusts that ``util.get_next_issue_time`` starting from the current issue time (+ns) always returns a greater timestamp (it does) then the loop will exit. 

Basically, I'm confident the loop will exit, and the workers running the jobs have a finite lifetime per job, so we'll be able to spot issues.

Comment (wholmgren) [reply solarforecastarbiter/reference_forecasts/main.py:566]: works for me

File: solarforecastarbiter/io/reference_observations/srml.py [hunk 14]
```diff
@@ -209,8 +209,9 @@ def initialize_site_forecasts(api, site):
     site : datamodel.Site
         The site object for which to create Forecasts.
     """
-    common.create_forecasts(api, site, srml_variable_map.values(),
-                            default_forecasts.TEMPLATE_DETERMINISTIC_FORECASTS)
+    common.create_forecasts(
+        api, site, srml_variable_map.values(),
+        default_forecasts.TEMPLATE_DETERMINISTIC_NWP_FORECASTS)
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/srml.py:214]: SRML now has a few PV systems, so I think we should support the full reference forecast suite for the network.

Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/srml.py:214]: Also noting that this line and the corresponding one in `crn.py` are not tested. OK to address that separately.

File: solarforecastarbiter/io/reference_observations/default_forecasts.py [hunk 15]
```diff
@@ -112,5 +112,71 @@ def is_in_nwp_domain(site):
...
+        lead_time_to_start=pd.Timedelta('1h'),
+        interval_length=pd.Timedelta('1h'),
+        run_length=pd.Timedelta('1h'),
+        interval_label='beginning',
+        interval_value_type='interval_mean',
+        variable='ghi',
+        site=_DUMMY_SITE,
+        extra_parameters=json.dumps(
+            {'is_reference_persistence_forecast': True,
+             'index_persistence': False})
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/default_forecasts.py:142]: I should have noticed this the first time through... why default to index=False? I think index=True would be more interesting and useful to most users.

Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/default_forecasts.py:142]: I'd be ok with both but I'm guessing you'd rather pick one.

File: docs/source/whatsnew/1.0.0rc1.rst [hunk 16]
```diff
@@ -16,6 +16,15 @@ API Changes
   with units of % of DC capacity/C instead of 1/C (:issue:`350`) (:pull:`433`)
 * Add :py:func:`solarforecastarbiter.io.utils.mock_raw_report_endpoints` to mock
   report API endpoints used, primarily to test report serialization (:pull:`441`)
+* :py:func:`solarforecastarbiter.io.reference_data.common.create_forecasts` now
+  calls :py:func:`solarforecastarbiter.io.reference_data.common.create_persistence_forecasts`
+  and :py:func:`solarforecastarbiter.io.reference_data.common.create_nwp_forecasts`
+  which create the Forecast objects from persistence and NWP templates,
+  respectively. (:issue:`375`) (:pull:`455`)
+* Added :py:func:`solarforecastarbiter.reference_forecasts.main.make_latest_persistence_forecasts`
+  to support periodic generation of persistence reference forecasts (:issue:`375`)
```
Comment (wholmgren) [inline docs/source/whatsnew/1.0.0rc1.rst:25]:
```suggestion
  to support periodic generation of persistence reference forecasts. (:issue:`375`)
```

File: solarforecastarbiter/io/reference_observations/common.py [hunk 17]
```diff
@@ -535,3 +541,83 @@ def create_forecasts(api, site, variables, templates):
...
+
+    Returns
+    -------
+    list
+        A list of all Forecast/ProbabilisticForecast objects created
+
+    See Also
+    --------
+    :py:func:`solarforecastarbiter.io.reference_data.common.create_nwp_forecasts
+    :py:func:`solarforecastarbiter.io.reference_data.common.create_persistence_forecasts
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/common.py:610]:
```suggestion
    solarforecastarbiter.io.reference_data.common.create_nwp_forecasts
    solarforecastarbiter.io.reference_data.common.create_persistence_forecasts
```

File: solarforecastarbiter/reference_forecasts/main.py [hunk 18]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
         logger.info('No forecasts to be made at %s', run_time)
         return
     process_nwp_forecast_groups(session, run_time, execute_for)
+
+
+def _is_reference_persistence_forecast(extra_params_string):
+    match = re.search('is_reference_persistence_forecast(["\\s\\:]*)true',
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:471]: not an r string?

File: solarforecastarbiter/reference_forecasts/main.py [hunk 19]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+                      extra_params_string, re.I)
+    return match is not None
+
+
+def generate_reference_persistence_forecast_parameters(
+        session, forecasts, observations, max_run_time):
+    """Sort through all *forecasts* to find those that should be
+    generated by the Arbiter from persisting Observation values. The
+    forecast must have a is_reference_persistence_forecast=true and an
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:480]: if it's supposed to be a JSON string then I'd expect `'is_reference_persistence_forecast': true`

File: solarforecastarbiter/reference_forecasts/main.py [hunk 20]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+                'Failed to decode extra_parameters for %s: %s as JSON',
+                fx.name, fx.forecast_id)
+            continue
+
+        try:
+            observation_id = extra_parameters['observation_id']
+        except KeyError:
+            logger.error(
+                'Forecast, %s: %s, has no observation_id to base forecasts'
+                ' off of. Cannot make forecast.',
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:531]:
```suggestion
                ' off of. Cannot make persistence forecast.',
```

File: solarforecastarbiter/reference_forecasts/main.py [hunk 21]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+                observation_id, fx.name, fx.forecast_id)
+            continue
+        observation = observation_dict[observation_id]
+
+        index = extra_parameters.get('index_persistence', False)
+        obs_mint, obs_maxt = session.get_observation_time_range(observation_id)
+        if pd.isna(obs_maxt):  # no observations to use anyway
+            logger.info(
+                'No observation values to use for %s:%s from observation %s',
+                fx.name, fx.forecast_id, observation_id)
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:547]: consistent formatting for `'...%s:%s...', fx.name, fx.forecast_id` might make searching easier later on. The error messages above have a mix of space/no-space and sentence position.

fine to keep it this way though.

File: solarforecastarbiter/reference_forecasts/main.py [hunk 22]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+            continue
+
+        fx_mint, fx_maxt = session.get_forecast_time_range(fx.forecast_id)
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:550]: what happens if there's a time span in which forecasts are not created because of something like delayed data ingest (e.g. the SURFRAD problems) or a crashed worker and then they start again? Is it possible to fill that gap?

File: solarforecastarbiter/reference_forecasts/tests/test_main.py [hunk 23]
```diff
@@ -534,3 +537,375 @@ def test_make_latest_nwp_forecasts(forecast_list, mocker, issue_buffer, empty):
...
+        ),
+    ]
+
+
+def test_generate_reference_persistence_forecast_parameters_multiple(
+        mocker, perst_fx_obs):
+    forecasts, observations = perst_fx_obs
+    forecasts[0] = forecasts[0].replace(
+        extra_parameters=(forecasts[0].extra_parameters[:-1] +
+                          ', "index_persistence": true}')
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/tests/test_main.py:841]: Do we need a test that asserts that a persistence index function is called when `"index_persistence": true`?

File: solarforecastarbiter/reference_forecasts/main.py [hunk 24]
```diff
@@ -465,3 +465,143 @@ def make_latest_nwp_forecasts(token, run_time, issue_buffer, base_url=None):
...
+            continue
+        if observation_id not in observation_dict:
+            logger.error(
+                'Observation %s not in set of given observations.'
+                ' Cannot generate persistence forecast for %s:%s.',
+                observation_id, fx.name, fx.forecast_id)
+            continue
+        observation = observation_dict[observation_id]
+
+        index = extra_parameters.get('index_persistence', False)
```
Comment (wholmgren) [inline solarforecastarbiter/reference_forecasts/main.py:542]: do we need a check like above that it's ok to make index persistence forecasts for the variable?

File: solarforecastarbiter/io/reference_observations/srml.py (hunk 14 above)
Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/srml.py:214]: it'll add 20-30% more forecasts to make, but should be ok

File: solarforecastarbiter/reference_forecasts/main.py (hunk 18 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:471]: forgot about them actually

File: solarforecastarbiter/reference_forecasts/main.py (hunk 22 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:550]: that's actually the whole point of looking at ``fx_maxt``. We find the next issue time based on it to go from ``fx_maxt`` to ``fx_maxt + run_length`` and then continue up to some max time. In the while loop, we also make sure all the data required for a forecast is available before making it.

For data gaps, we don't go back and make the persistence forecast if the data is filled in, but I think that should just be dealt with separately

File: solarforecastarbiter/io/reference_observations/default_forecasts.py (hunk 15 above)
Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/default_forecasts.py:142]: I didn't want to fuss with adjusting it to true/false based on the variable since  only works for power/irradiance. It would actually be easy to do though

File: solarforecastarbiter/io/reference_observations/default_forecasts.py (hunk 13 above)
Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/default_forecasts.py:115]: I guess I made the run length 1h, is that what you meant or 1day?

File: solarforecastarbiter/reference_forecasts/tests/test_main.py (hunk 23 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/tests/test_main.py:841]: we'll leave that up to the ``run_persistence`` function and just make sure the value of index is correct when it is called

File: solarforecastarbiter/reference_forecasts/main.py (hunk 24 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:542]: eh, it should be properly specified on creation and this only applies to reference forecasts anyway. We could add a function that checks if an index persistence fx is allowed for a given Forecast, but I'll leave that for another issue/pr

File: solarforecastarbiter/reference_forecasts/main.py (hunk 22 above)
Comment (wholmgren) [reply solarforecastarbiter/reference_forecasts/main.py:550]: My only concern is that we have the ability to easily deal with gaps once they are created and I'm wondering if this function should be modified to support that.

File: solarforecastarbiter/io/reference_observations/default_forecasts.py (hunk 13 above)
Comment (wholmgren) [reply solarforecastarbiter/io/reference_observations/default_forecasts.py:115]: I meant 1 day, as in line 30 of the workshop repo example: https://nbviewer.jupyter.org/github/SolarArbiter/workshop/blob/master/reference_forecasts.ipynb#persistence-forecasts

File: solarforecastarbiter/reference_forecasts/main.py (hunk 22 above)
Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:550]: I think that is a separate issue and interface. This already handles the case where data is automatically fetched although there may be a delay or gaps. To fill in the data gaps, we already need to manually run a cli script, so it would be reasonable to fill in the persistence fx gaps in the same way.

We also don't want this to try and automatically fill in the forecast gaps everytime it runs. There was likely a reason the gaps exist and we would likely hit the same error over and over.

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:550]: well adding in a cli function to run this and possible specifying ``--start`` like ``referencenwp`` and ``referencedata`` wouldn't pollute this PR too much. But we should also be able fetch the forecast, find missing periods, and make those missing series from a CLI script, which I think belongs in a new pr

Comment (wholmgren) [reply solarforecastarbiter/reference_forecasts/main.py:550]: Ok that makes sense.

Comment (alorenzo175) [reply solarforecastarbiter/reference_forecasts/main.py:550]: https://github.com/SolarArbiter/solarforecastarbiter-core/issues/459

**Comment by alorenzo175** [comment]
> any concern about extending this code for those cases? Looks ok to me.

Nope, as far as I know it should be straightforward.
